src/machine.rs

```diff
@@ -423,7 +423,7 @@ impl<'mir, 'tcx> Evaluator<'mir, 'tcx> {
     /// Sets up the "extern statics" for this machine.
     fn init_extern_statics(this: &mut MiriEvalContext<'mir, 'tcx>) -> InterpResult<'tcx> {
         match this.tcx.sess.target.os.as_ref() {
-            "linux" => {
+            "linux" | "freebsd" => {
```

Comment:
I don't think we want *all* the Linux vars on FreeBSD. Some are probably very Linux-specific, like `__cxa_thread_atexit_impl`.

So please start a new section just for freebsd.

Reply:
> I don't think we want _all_ the Linux vars on FreeBSD. Some are probably very Linux-specific, like `__cxa_thread_atexit_impl`.
> 
> So please start a new section just for freebsd.

`__cxa_thread_atexit_impl` seems to exist on freebsd
it looks like it was introduced by `https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=215709` 
(https://github.com/freebsd/freebsd-src/blob/main/lib/libc/stdlib/Symbol.map#L118)

Reply:
Still, please don't add it until a test demands it.

Reply:
Removed, we're left with getrandom now

Reply:
I meant to reduce it all the way to environ. :) That's the only thing we are actually testing now.

Reply:
like so ?
```
            "freebsd" => {
                // "environ"
                Self::add_extern_static(
                    this,
                    "environ",
                    this.machine.env_vars.environ.unwrap().ptr,
                );
                // A couple zero-initialized pointer-sized extern statics.
                // Most of them are for weak symbols, which we all set to null (indicating that the
                // symbol is not supported, and triggering fallback code which ends up calling a
                // syscall that we do support).
                {
                    let name = &"environ";
                    let layout = this.machine.layouts.usize;
                    let place = this.allocate(layout, MiriMemoryKind::ExternStatic.into())?;
                    this.write_scalar(Scalar::from_machine_usize(0, this), &place.into())?;
                    Self::add_extern_static(this, name, place.ptr);
                }
```